File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.cs
public class Solution {
    public int WaysToSplitArray(int[] nums) {
        long rsum = 0;
        long lsum = 0;
        foreach (var num in nums)
            rsum += num;
        
        int cnt = 0;
        for (var i = 0; i < nums.Length - 1; i++) {
            lsum += nums[i];
            rsum -= nums[i];

            if (lsum >= rsum)
                cnt++;
        }
        return cnt;
    }
}

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.c
int waysToSplitArray(int* nums, int numsSize) {
    long long int rsum = 0, lsum = 0;
    for (int i = 0; i < numsSize; i++)
        rsum += nums[i];
    int cn = 0;
    for (int i = 0; i < numsSize - 1; i++) {
        lsum += nums[i];
        rsum -= nums[i];
        if (lsum >= rsum)
            cn++;
    }
    return cn;
}